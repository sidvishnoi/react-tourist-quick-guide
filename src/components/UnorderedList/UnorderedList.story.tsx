import { array, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import UnorderedList from '.';

const stories = storiesOf('UnorderedList', module);
stories.addDecorator(withKnobs);

stories.add('with plain text items', () => {
  const items = array('items', ['One', 'Two', 'Three']);
  const listStyleType = text('list-style-type', 'square');

  return <UnorderedList items={items} style={{ listStyleType }} />;
});
