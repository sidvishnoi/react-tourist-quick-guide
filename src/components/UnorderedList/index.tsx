import * as React from 'react';
import styled from 'styled-components';

import ListItem from '../ListItem';

const UnorderedList = styled.ul`
  margin: 0.5rem 0 0 0.5rem;
`;

export interface UnorderedListProps {
  items?: React.ReactNode[];
  children?: React.ReactNode[];
  [prop: string]: any;
}

export default (props: UnorderedListProps) => {
  const { items, children, ...rest } = props;
  const listItems = items ? items : children;
  if (!listItems || !Array.isArray(listItems)) {
    return <></>;
  }
  return (
    <UnorderedList {...rest}>
      {listItems.map((item, i) => (
        <ListItem key={i}>{item}</ListItem>
      ))}
    </UnorderedList>
  );
};
