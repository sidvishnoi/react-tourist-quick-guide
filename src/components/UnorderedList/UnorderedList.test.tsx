import React = require('react');
import * as renderer from 'react-test-renderer';

import UnorderedList from '.';

describe('Component - UnorderedList', () => {
  it('renders nothing if no items or children provided', () => {
    const tree = renderer.create(<UnorderedList />);
    expect(tree).toMatchSnapshot();
  });

  it('renders empty list if items is empty array', () => {
    const tree = renderer.create(<UnorderedList items={[]} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders list with plain text items', () => {
    const items = ['one', 'two'];
    const tree = renderer.create(<UnorderedList items={items} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders list with JSX items', () => {
    const items = [<span>one</span>, <span>two</span>];
    const tree = renderer.create(<UnorderedList items={items} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders JSX children', () => {
    const tree = renderer.create(
      <UnorderedList>
        <span>one</span>
        <span>two</span>
      </UnorderedList>,
    );
    expect(tree).toMatchSnapshot();
  });
});
