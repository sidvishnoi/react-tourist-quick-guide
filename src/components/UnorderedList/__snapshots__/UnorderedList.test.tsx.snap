// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - UnorderedList renders JSX children 1`] = `
<ul
  className="sc-bwzfXH gjGEUQ"
>
  <li
    className="sc-bdVaJa DYeZc"
  >
    <span>
      one
    </span>
  </li>
  <li
    className="sc-bdVaJa DYeZc"
  >
    <span>
      two
    </span>
  </li>
</ul>
`;

exports[`Component - UnorderedList renders empty list if items is empty array 1`] = `
<ul
  className="sc-bwzfXH gjGEUQ"
/>
`;

exports[`Component - UnorderedList renders list with JSX items 1`] = `
<ul
  className="sc-bwzfXH gjGEUQ"
>
  <li
    className="sc-bdVaJa DYeZc"
  >
    <span>
      one
    </span>
  </li>
  <li
    className="sc-bdVaJa DYeZc"
  >
    <span>
      two
    </span>
  </li>
</ul>
`;

exports[`Component - UnorderedList renders list with plain text items 1`] = `
<ul
  className="sc-bwzfXH gjGEUQ"
>
  <li
    className="sc-bdVaJa DYeZc"
  >
    one
  </li>
  <li
    className="sc-bdVaJa DYeZc"
  >
    two
  </li>
</ul>
`;

exports[`Component - UnorderedList renders nothing if no items or children provided 1`] = `null`;
